### PR TITLE
Np 7142 person rename query to name

### PR DESCRIPTION
**Pull request by svenngjeto**

### CODE REVIEW

File: cristin-commons/src/main/java/no/unit/nva/cristin/common/ErrorMessages.java [hunk 1]
```diff
@@ -41,6 +44,8 @@
             + "application/ld+json";
     public static final String ERROR_MESSAGE_INVALID_QUERY_PARAMS_ON_SEARCH =
         "Invalid query param supplied. Valid ones are 'query', 'page', 'results' and 'language'";
+    public static final String ERROR_MESSAGE_INVALID_QUERY_PARAMS_ON_PERSON_SEARCH =
+            "Invalid query param supplied. Valid ones are 'name', 'page', 'results' and 'language'";
```
Comment (brinxmat) [inline cristin-commons/src/main/java/no/unit/nva/cristin/common/ErrorMessages.java:48]: param -> parameter, ones -> parameters

Comment (brinxmat) [reply cristin-commons/src/main/java/no/unit/nva/cristin/common/ErrorMessages.java:48]: Also, generally change this in the file

File: cristin-commons/src/main/java/no/unit/nva/cristin/common/handler/CristinQueryHandler.java [hunk 2]
```diff
@@ -57,6 +59,14 @@ protected String getValidQuery(RequestInfo requestInfo) throws BadRequestExcepti
                 .orElseThrow(() -> new BadRequestException(ERROR_MESSAGE_QUERY_MISSING_OR_HAS_ILLEGAL_CHARACTERS));
     }
 
+    protected String getValidName(RequestInfo requestInfo) throws BadRequestException {
+        return getQueryParam(requestInfo, NAME)
```
Comment (brinxmat) [inline cristin-commons/src/main/java/no/unit/nva/cristin/common/handler/CristinQueryHandler.java:63]: Parameter

File: cristin-person/src/main/java/no/unit/nva/cristin/person/handler/QueryCristinPersonHandler.java [hunk 3]
```diff
@@ -1,24 +1,31 @@
...
+import static no.unit.nva.cristin.common.ErrorMessages.ERROR_MESSAGE_INVALID_QUERY_PARAMS_ON_PERSON_SEARCH;
+import static no.unit.nva.cristin.model.JsonPropertyNames.NAME;
+import static no.unit.nva.cristin.model.JsonPropertyNames.NUMBER_OF_RESULTS;
+import static no.unit.nva.cristin.model.JsonPropertyNames.PAGE;
+
 @SuppressWarnings("unused")
 public class QueryCristinPersonHandler extends CristinQueryHandler<Void, SearchResponse<Person>> {
 
+
+    private static final Set<String> VALID_QUERY_PARAMS = Set.of(NAME, PAGE, NUMBER_OF_RESULTS);
```
Comment (brinxmat) [inline cristin-person/src/main/java/no/unit/nva/cristin/person/handler/QueryCristinPersonHandler.java:28]: PARAMS -> PARAMETERS

File: cristin-person/src/main/java/no/unit/nva/cristin/person/handler/QueryCristinPersonHandler.java [hunk 4]
```diff
@@ -42,11 +49,11 @@ public QueryCristinPersonHandler(CristinPersonApiClient apiClient, Environment e
 
         validateQueryParamKeys(requestInfo);
 
-        String query = getValidQuery(requestInfo); // TODO: Discuss validation of name query
+        String name = getValidName(requestInfo);
         String page = getValidPage(requestInfo);
         String numberOfResults = getValidNumberOfResults(requestInfo);
 
-        Map<String, String> requestQueryParams = buildParamMap(query, page, numberOfResults);
+        Map<String, String> requestQueryParams = buildParamMap(name, page, numberOfResults);
```
Comment (brinxmat) [inline cristin-person/src/main/java/no/unit/nva/cristin/person/handler/QueryCristinPersonHandler.java:56]: De-abbreviate params generally

**Review by brinxmat — APPROVED** [review]

**Review by torarnet — DISMISSED** [review]

File: cristin-project/src/main/java/no/unit/nva/cristin/projects/FetchCristinProjects.java [hunk 5]
```diff
@@ -59,9 +61,9 @@ protected FetchCristinProjects(CristinApiClient cristinApiClient, Environment en
     }
 
     @Override
-    protected void validateQueryParamKeys(RequestInfo requestInfo) throws BadRequestException {
-        if (!VALID_QUERY_PARAMS.containsAll(requestInfo.getQueryParameters().keySet())) {
-            throw new BadRequestException(ERROR_MESSAGE_INVALID_QUERY_PARAMS_ON_SEARCH);
+    protected void validateQueryParameterKeys(RequestInfo requestInfo) throws BadRequestException {
+        if (!VALID_QUERY_PARAMETERS.containsAll(requestInfo.getQueryParameters().keySet())) {
+            throw new BadRequestException(ERROR_MESSAGE_INVALID_QUERY_PARAMETERS_ON_SEARCH);
```
Comment (brinxmat) [inline cristin-project/src/main/java/no/unit/nva/cristin/projects/FetchCristinProjects.java:66]: In a new PR, can you ensure that we replace this string with a template string, where we input the set of valid parameters — in all places that we do this?

**Review by brinxmat — APPROVED** [review]

**Pull request merged**